File: javascript/VParabola.js

function VParabola(s)
{
	this.cEvent = null;
	this.parent = null;
	this._left = null;
	this._right = null;
	

	this.site = s;
	this.isLeaf = (this.site != null);
}

VParabola.prototype = {
    get left(){
        return this._left;
    },
    get right(){
        return this._right;
    },
	
	set left(p){
        this._left = p;
		p.parent = this;
    },
    set right(p){
        this._right = p;
		p.parent = this;
    }
};

File: ported/VParabola.py
__author__ = 'sean'

class VParabola:
    def __init__(self, s):
        self.cEvent = None
        self.parent = None
        self._left = None
        self._right = None

        self.site = s
        self.isLeaf = (self.site != None)
    #TODO:  need to check the set & get calls
    def getLeft(self):
        return self._left
    def getRight(self):
        return eslf._right
    def setLeft(self, p):
        self._left = p
        p.parent = self
    def setRight(self,p):
        self._right = p
        p.parent = self